### PR TITLE
[TEST] Increase code coverage with easy test additions

### PR DESCRIPTION
**Pull request by smehringer**
Add error checkings to multiple tests

### CODE REVIEW

File: include/chopper/pack/aggregate_by.hpp [hunk 1]
```diff
@@ -39,14 +39,14 @@ inline void sort_by(pack_data & data, size_t column_to_sort_by)
 
 inline void aggregate_by(pack_data & data, size_t column_to_aggregate_by)
 {
+    if (data.filenames.size() == 0)
```
Comment (eseiler) [inline include/chopper/pack/aggregate_by.hpp:42]:
```suggestion
    if (data.filenames.empty())
```

File: test/api/pack/aggregate_by_test.cpp [hunk 2]
```diff
@@ -36,6 +36,12 @@ TEST(sort_by_test, small_example)
     }
 }
 
+TEST(aggregate_by_test, filenames_are_emprty)
+{
+    pack_data data;
+    aggregate_by(data, 0);
+}
```
Comment (eseiler) [inline test/api/pack/aggregate_by_test.cpp:43]: It shouldn't change anything, right?
So maybe:

```suggestion
{
    pack_data data{};

    char original_binary[sizeof(data)];
    std::memcpy(original_binary, &data, sizeof(data));
    
    aggregate_by(data, 0);
    
    char new_binary[sizeof(data)];
    std::memcpy(new_binary, &data, sizeof(data));

    EXPECT_TRUE(std::strcmp(original_binary, new_binary) == 0);
}
```

Or implement the comparison operator :D

File: test/data/filenames_and_counts.tsv [hunk 3]
```diff
@@ -1,3 +1,4 @@
 file1	1000
 file2	2000
+
```
Comment (eseiler) [inline test/data/filenames_and_counts.tsv:3]: Why this?

This reminds me that we should probably skip lines only containing `\n` when parsing files.

**Review by eseiler — CHANGES_REQUESTED** [review]

File: test/api/pack/aggregate_by_test.cpp (hunk 2 above)
Comment (smehringer) [reply test/api/pack/aggregate_by_test.cpp:43]: Yes it does change data. Aggregation means I change the data rows by merging them.
E.g. aggregate by column 2 (specs)
```
file1 spec1 100
file2 spec1 50
file3 spec2 60
``` 
yields
```
file1;file2 spec1 150
file3 spec2 60
```

File: test/data/filenames_and_counts.tsv (hunk 3 above)
Comment (smehringer) [reply test/data/filenames_and_counts.tsv:3]: > This reminds me that we should probably skip lines only containing \n when parsing files.

That's why. I skip lines containing only `\n` and now I also test it.

File: test/api/pack/aggregate_by_test.cpp (hunk 2 above)
Comment (eseiler) [reply test/api/pack/aggregate_by_test.cpp:43]: But, `data` is empty here, so it does nothing, right?
Shouldn't we confirm that it does nothing?

**Review by eseiler — APPROVED** [review]

**Pull request merged**